src/s2/shape.rs

```diff
@@ -0,0 +1,251 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use r2::point::Point;
+use s2::point::Point as s2Point;
+use std::cmp::Ordering;
+
+// Edge represents a geodesic edge consisting of two vertices. Zero-length edges are
+// allowed, and can be used to represent points.
+
+pub struct Edge {
+	v0: Point, 
+	v1: Point
+}
+
+
+impl Eq for Edge {}
+
+impl Ord for Edge {
+	fn cmp(&self, other: &Edge) -> Ordering {
+		let v0cmp = self.v0.cmp(&other.v0);
+		if v0cmp != Ordering::Equal {
+			v0cmp
+		} else {
+			self.v0.cmp(&other.v1)
+		}
+	}
+}
+
+impl PartialOrd for Edge {
+    fn partial_cmp(&self, other: &Edge) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl PartialEq for Edge {
+    fn eq(&self, other: &Edge) -> bool {
+        self.cmp(other) == Ordering::Equal
+    }
+}
+
+
+// sortEdges sorts the slice of Edges in place.
+pub fn sort_edges(mut e: Vec<Edge>) {
```

Comment:
How about `pub fn sort_edges(e: &mut [Edge])`? The function takes `Vec<Edge>` and drops the vector after sort, so there's no way to get a sorted result.